pkg/apis/extensions/validation/dnsrecord.go

```diff
@@ -58,7 +58,8 @@ func ValidateDNSRecordSpec(spec *extensionsv1alpha1.DNSRecordSpec, fldPath *fiel
 	}
 
 	// This will return FieldValueRequired for an empty spec.Name
-	allErrs = append(allErrs, validation.IsFullyQualifiedDomainName(fldPath.Child("name"), strings.TrimPrefix(spec.Name, "*."))...)
+	// allow leading '_' as used for DNS challenges (e.g. Let's Encrypt)
+	allErrs = append(allErrs, validation.IsFullyQualifiedDomainName(fldPath.Child("name"), strings.TrimPrefix(strings.TrimPrefix(spec.Name, "_"), "*."))...)
```

Comment:
We could trim `_` if `RecordType` is `extensionsv1alpha1.DNSRecordTypeTXT` only. `A` or `AAAA` records should still not start with `_`, shouldn't they?
WDYT?

Reply:
I think it is a grey zone anyway, but I have changed it.